test/e2e/runtime_knative.go

```diff
@@ -33,38 +41,118 @@ func RuntimeKnativeTest(t *testing.T) {
+	// create application deployment and wait
+	err := f.Client.Create(goctx.TODO(), exampleRuntime, &framework.CleanupOptions{TestContext: ctx, Timeout: time.Second, RetryInterval: time.Second})
 	if err != nil {
 		util.FailureCleanup(t, f, namespace, err)
 	}
 
-	err = util.WaitForKnativeDeployment(t, f, namespace, applicationName, retryInterval, timeout)
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, applicationName, 1, retryInterval, operatorTimeout)
+	if err != nil {
+		util.FailureCleanup(t, f, namespace, err)
+	}
+
+	// Knative service should not be deployed.
+	isDeployed, err := util.IsKnativeServiceDeployed(t, f, namespace, applicationName)
 	if err != nil {
 		util.FailureCleanup(t, f, namespace, err)
 	}
+	if isDeployed {
+		util.FailureCleanup(t, f, namespace, 
+			errors.New("knative service is deployed when CreateKnativeService is set to false"))
+	}
 }
 
-func isKnativeInstalled(t *testing.T, f *framework.Framework) bool {
-	ns := &corev1.NamespaceList{}
-	err := f.Client.List(goctx.TODO(), ns)
+func testKnIsTrueAndTurnOff(t *testing.T, f *framework.Framework, ctx *framework.TestCtx, namespace string) {
+	knativeBool := true
+	applicationName := "example-runtime-knative-1"
+	exampleRuntime := util.MakeBasicRuntimeComponent(t, f, applicationName, namespace, 1)
+	exampleRuntime.Spec.CreateKnativeService = &knativeBool
+
+	// create application deployment and wait
+	err := f.Client.Create(goctx.TODO(), exampleRuntime, &framework.CleanupOptions{TestContext: ctx, Timeout: time.Second, RetryInterval: time.Second})
 	if err != nil {
-		t.Fatalf("Error occurred while trying to find knative-serving %v", err)
+		util.FailureCleanup(t, f, namespace, err)
 	}
-	for _, val := range ns.Items {
-		if val.Name == "knative-serving" {
-			return true
+
+	err = util.WaitForKnativeDeployment(t, f, namespace, applicationName, retryInterval, timeout)
+	if err != nil {
+		util.FailureCleanup(t, f, namespace, err)
+	}
+
+	// If deployment not cleared, test fails.
+	dep := &appsv1.Deployment{}
+	err = f.Client.Get(goctx.TODO(), types.NamespacedName{Name: applicationName, Namespace: namespace}, dep)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			util.FailureCleanup(t, f, namespace, err)
 		}
 	}
-	return false
+
+	// turn the runtime component off / set CreateKnativeService to false.
+	target := types.NamespacedName{Name: applicationName, Namespace: namespace}
+	err = util.UpdateApplication(f, target, func(r *appstacksv1beta1.RuntimeComponent) {
+		knativeBool = false
+		r.Spec.CreateKnativeService = &knativeBool
+	})
+	if err != nil {
+		util.FailureCleanup(t, f, namespace, err)
+	}
+
+	// wait for the change to take effect and verify the state
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, applicationName, 1, retryInterval, operatorTimeout)
+	if err != nil {
+		util.FailureCleanup(t, f, namespace, err)
+	}
+	dep = &appsv1.Deployment{}
```

Comment:
The `f.Client.Get(...)` call here can be removed, as you wait for the same deployment in the statements above.